src/lsm/grid.zig

```diff
@@ -448,22 +509,14 @@ pub fn GridType(comptime Storage: type) type {
             const cache_block = &grid.cache_blocks[cache_index];
             std.mem.swap(BlockPtr, cache_block, completed_write.block);
             std.mem.set(u8, completed_write.block.*, 0);
```

Comment:
Unrelated to this PR, but why are we paying for memory bandwidth over a whole 64 KiB block by zeroing the whole block here? Is the intention to zero the whole block or only to zero the header?

Reply:
The intention is indeed to zero the whole block. We could certainly optimize this later, but right now `TableBuilder` expects any blocks it gets to be zeroed already.

(The motivation for zeroing the whole block was to make it easy to detect memory corruption + dangerously-reused blocks. Both of those issues are normally [very hard to track down](https://www.scattered-thoughts.net/log/0032/#hash-log)).

Reply:
Thanks for the pointer to `TableBuilder`'s expectations w.r.t. padding. I was aware of the cache race bug but not that padding was now also coupled implicitly to Grid, rather than padding being set explicitly in TableBuilder.

The motivation behind the cache here in the first place, is to use Direct I/O, not only for safety to handle fsync failure correcty, but also for performance to avoid the memcopy in the write path to the kernel's page cache. However, this memset effectively reintroduces the memcopy.

This is important, since memcopy in the data plane uses CPU and thrashes the L1/L2/L3 cache. This is the reason we've been so careful everywhere to do zero-copy (`recv()` into message buffer, check checksum, Direct I/O `write()` from message buffer, `send()` from message buffer).

Reply:
For example, `TableBuilder` should take responsibility to explicitly zero padding, and then the tests that @chaitanyabhandari is about to do for `TableBuilder` could then test this exhaustively.

Here, in Grid, we could then rather zero the block header, if that works/helps, so that the safety check impacts the control plane rather than the data plane.

This can be a separate PR, and @chaitanyabhandari, it would be great if you can tackle this.